### DISCUSSION TITLE
Max file size on medias

### DISCUSSION THREAD

**Discussion by michaeljcoyne** [area17/twill#1631] [General]
What is the Twill way of validating the max file size on the medias form field?

In this case it is on a module.

I have a user completely oblivious of any optimisation concepts.

**Comment by haringsrob** [comment]
Hey @michaeljcoyne,

The limit is set by https://github.com/area17/twill/blob/308d4e393de6ff5b05c36f9b69facfc113ae8cbd/config/media-library.php#L28

In addition to that there's your servers max file sizes.

https://www.a2hosting.com/kb/developer-corner/php/using-php.ini-directives/php-maximum-upload-file-size

Is this what you were looking for?

**Reply by michaeljcoyne** [reply]
Hi @haringsrob 

Not really - I would like more granular control on the validation on a block level. As some images I want to allow upto 2mb but for list images only like 100k.

So ideally hooking into the request validation per block / module rather than one size fits all.

Thanks

Michael

**Reply by kallefrombosnia** [reply]
Just a thought, this goes trough `ModuleRequest` firstly, there should be set upload limit check on the media field?
Im not sure for the blocks.
